### PR TITLE
Add caldav third-party stubs

### PR DESCRIPTION
**Pull request by srittau**

### CODE REVIEW

File: stubs/caldav/caldav/davclient.pyi [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+        self,
+        url: str,
+        proxy: str | None = ...,
+        username: str | None = ...,
+        password: str | None = ...,
+        auth: AuthBase | None = ...,
+        ssl_verify_cert: bool | str = ...,
+        ssl_cert: str | tuple[str, str] | None = ...,
+    ) -> None: ...
+    def principal(self, url: str | ParseResult | SplitResult | URL | None) -> Principal: ...
```
Comment (JelleZijlstra) [inline stubs/caldav/caldav/davclient.pyi:48]:
```suggestion
    def principal(self, *, url: str | ParseResult | SplitResult | URL | None) -> Principal: ...
```
It does `Principal(client=self, *largs, **kwargs)`, which I don't think can actually work when largs is non-empty.

File: stubs/caldav/caldav/lib/url.pyi [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+
+class URL:
+    def __init__(self, url: str | ParseResult | SplitResult) -> None: ...
+    def __bool__(self) -> bool: ...
+    def __ne__(self, other: object) -> bool: ...
+    def __eq__(self, other: object) -> bool: ...
+    def __hash__(self) -> int: ...
+    @overload
+    @classmethod
+    def objectify(cls, url: None) -> URL | None: ...
```
Comment (JelleZijlstra) [inline stubs/caldav/caldav/lib/url.pyi:12]:
```suggestion
    def objectify(cls, url: None) -> None: ...
```
It always returns None.

**Review by JelleZijlstra — COMMENTED** [review]
Didn't look at objects.pyi yet

File: stubs/caldav/caldav/davclient.pyi (hunk 1 above)
Comment (srittau) [reply stubs/caldav/caldav/davclient.pyi:48]: I'm not sure what you mean. Can you give an example?

Comment (JelleZijlstra) [reply stubs/caldav/caldav/davclient.pyi:48]: The definition of this method has `*largs, **kwargs`, but the only thing it does with its args and kwargs is `Principal(client=self, *largs, **kwargs)`. The signature of Principal is `(client, url=None)`, so the only legal ways to call this method are `.principal()` and `.principal(url=url)`.

So in addition to what I wrote above there should be an `= ...`.

File: stubs/caldav/caldav/objects.pyi [hunk 3]
```diff
@@ -0,0 +1,174 @@
...
+from urllib.parse import ParseResult, SplitResult
+
+from .davclient import DAVClient
+from .elements.cdav import CompFilter, ScheduleInboxURL, ScheduleOutboxURL
+from .lib.url import URL
+
+_CC = TypeVar("_CC", bound=CalendarObjectResource)
+
+VBase = Any  # actually vobject.base.VBase
+vCalAddress = Any  # actually icalendar.vCalAddress
```
Comment (JelleZijlstra) [inline stubs/caldav/caldav/objects.pyi:15]:
```suggestion
_VBase = Any  # actually vobject.base.VBase
_vCalAddress = Any  # actually icalendar.vCalAddress
```
These don't exist at runtime

**Review by JelleZijlstra — CHANGES_REQUESTED** [review]
Just one other thing I noticed

**Review by JelleZijlstra — APPROVED** [review]

**Pull request merged**